Refactor things that run each tick into nicer tickable service interface

TraceLD opened the pull request.

TraceLD committed:
add ITickedService and convert old stuff to it

TraceLD committed:
remove redundant start()

john01dav reviewed (changes requested):
See my previous comment (it didn't register as a review in Github for some reason).

TraceLD committed:
(submit to John's twattery)^2

the absolute special child made me write this description: the service was renamed from ITickedService to TickableService upon his twatty request

TraceLD committed:
add TimedTickableService and refactor BroadcasterService

so that it uses it

john01dav commented on server/src/main/java/club/moddedminecraft/polychat/server/services/BroadcasterService.java:
```diff
@@ -6,39 +6,35 @@
     private int broadcastMsgsIndex;
 
-    public static final int BROADCAST_EVERY_X_IN_TICKS = (10 * 60 * 20);
+    public static final int INTERVAL_IN_TICKS = (10 * 60 * 20);
```
While we're here, there should probably be a comment saying that this is 10 mintues (at least, I think that it is, hence the need for a comment).

TraceLD replied:
I honestly have no clue. This interval was agreed on months ago and I don't remember.

132ikl replied:
10s*60s = 600s * 20t = 12000 ticks = 10 mins

john01dav commented on server/src/main/java/club/moddedminecraft/polychat/server/services/TimedTickableService.java:
```diff
@@ -0,0 +1,40 @@
+    /**
+     *  Triggered every {@link TimedTickableService#intervalInTicks} ticks.
+     */
+    public void onRun() {}
```
This should be `public abstract void onRun();`.

TraceLD committed:
make onRun() abstract
